clients/cobol-lsp-vscode-extension/src/services/reporter/TelemetryService.ts

```diff
@@ -0,0 +1,98 @@
+        return ((startTime && endTime) && (endTime >= startTime)) ? ((endTime - startTime) / 1000) : undefined;
+    }
+
+    /**
+     * This method collect the data event for telemetry purpose, apply basic validation
+     * and delegate the reporter service to sent them to the target telemetry server
+     * @param eventName name assigned for the event for the telemetry
+     * @param categories optional list of keywords (act as tags)
+     * @param notes optional brief description
+     * @param telemetryMeasurement optional set of numeric data with a key name
+     */
+    public static registerEvent(eventName: string, categories?: string[], notes?: string, telemetryMeasurement?: Map<string, number>) {
+        if (!this.isValidEventName(eventName)) {
+            return;
+        }
+        TelemetryFactory.getReporter().reportEvent(this.createTelemetryEvent(eventName, categories, notes, undefined, telemetryMeasurement));
+
+    }
+
+    /**
+     * This method collect the exception data event for telemetry purpose, apply basic validation
+     * and delegate the reporter service to sent them to the target telemetry server
+     * @param eventName name assigned for the event for the telemetry
+     * @param rootCause the stacktrace attached for debugging purpose.
+     * @param categories optional list of keywords (act as tags)
+     * @param notes optional brief description
+     * @param telemetryMeasurement optional set of numeric data with a key name
+     */
+    public static registerExceptionEvent(eventName = "RuntimeException", rootCause: string, categories?: string[], notes?: string, telemetryMeasurement?: Map<string, number>) {
+        if (!(this.isValidEventName(eventName) && this.isValidRootCause(rootCause))) {
+            return;
+        }
+
+        TelemetryFactory.getReporter().reportExceptionEvent(this.createTelemetryEvent(eventName, categories, notes, this.anonymizeContent(rootCause), telemetryMeasurement));
+    }
+
+    private static isValidEventName(eventName: string): boolean {
+        return eventName !== undefined && eventName.trim() !== "";
+    }
+
+    private static createTelemetryEvent(eventName: string, categories: string[], notes: string, rootCause?: string, telemetryMeasurement?: Map<string, number>) {
+        const telemetryEvent = new TelemetryEvent();
+        telemetryEvent.setEventName(eventName);
+        telemetryEvent.setCategories(this.resolveCategories(categories));
+        telemetryEvent.setNotes(notes);
+        telemetryEvent.setRootCause(rootCause);
+        telemetryEvent.setMeasurements(telemetryMeasurement);
+
+        return telemetryEvent;
+    }
+
+    private static resolveCategories(categories: string[]): string {
+        return (categories) ? categories.toString() : "N.D";
+    }
+
+    private static isValidRootCause(rootCause: string): boolean {
+        return rootCause !== undefined && rootCause.trim() !== "";
```

Comment:
Will this work if rootCause = null ?

Reply:
I will strongify it and provide unit test